api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -130,6 +137,28 @@ export class ScenariosController {
     );
   }
 
+  @ApiOperation({ description: 'Create feature set for scenario' })
+  @ApiCreatedResponse({ type: ScenarioResult })
+  @Post(':id/features/specification')
+  async createFeatureSetFor(
+    @Body(new ValidationPipe()) dto: CreateGeoFeatureSetDTO,
```

Comment:
@Dyostiq I've seen you giving advanced suggestions for Swagger documentation - this is a WIP, but your feedback would be appreciated: here and in the following endpoint we use DTOs that themselves use union types with class-transformer discriminators for nested validation, which from what I understand and see don't play well with the kind of introspectioni the Swagger plugin could do... short of not using union types and adding some duplication, would you have suggestions for the Swagger documentation to reflect the full shape of DTOs?

Reply:
Hmm, for unions/polymorphic type you may use `oneOf` (along with `getSchemaPath`) and `discriminator` property of `ApiProperty`

Reply:
(and `ApiExtraModels` in order to have the model referenced by `getSchemaPath` included in the open api doc)